app/modules/collaborations/models.py

```diff
@@ -17,13 +51,161 @@ class Collaboration(db.Model, HoustonModel):
+                )
+
+            collab_user_assoc = CollaborationUserAssociations(
+                collaboration=self, user=user_tup[0]
+            )
+
+            if (
+                approval_states is not None
+                and approval_states[guid_index] in CollaborationUserState.ALLOWED_STATES
+            ):
+                collab_user_assoc.read_approval_state = approval_states[guid_index]
+            if (
+                is_initiator is not None
+                and isinstance(is_initiator[guid_index], bool)
+                and is_initiator[guid_index] is True
+            ):
+                collab_user_assoc.initiator = is_initiator[guid_index]
+                # If you initiate you approve
+                log.warning(' setting APPROVED!')
+                collab_user_assoc.read_approval_state = CollaborationUserState.APPROVED
+
+            # with db.session.begin():
+            #     db.session.add(collab_user_assoc)
+            #     db.session.merge(user_tup[0])
+
+    def get_users(self):
+        users = []
+        for association in self.collaboration_user_associations:
+            users.append(association.user)
+        return users
+
+    def get_initiators(self):
+        users = []
+        for association in self.collaboration_user_associations:
+            if association.initiator is True:
+                users.append(association.user)
+        return users
+
+    def get_read_state(self):
+        read_state = None  # if you return this the collaboration is corrupt
+        for association in self.collaboration_user_associations:
+            if association.read_approval_state == CollaborationUserState.DECLINED:
+                # only one user need decline
+                read_state = CollaborationUserState.DECLINED
+                break
+            elif association.read_approval_state == CollaborationUserState.PENDING:
+                # only one user needs pending, but we should still keep looking for declined
+                read_state = CollaborationUserState.PENDING
+            elif (
+                association.read_approval_state == CollaborationUserState.APPROVED
+                and read_state is None
+            ):
+                # every association must approve to return approved, everything else overrides it
+                read_state = CollaborationUserState.APPROVED
+        return read_state
+
+    def get_edit_state(self):
+        edit_state = None
+        for association in self.collaboration_user_associations:
+            # you should never have a 'pending' and 'not_initiated' on the same collab, so one of either is enough
+            if association.edit_approval_state == CollaborationUserState.DECLINED:
+                edit_state = CollaborationUserState.DECLINED
+                break
+            elif association.edit_approval_state == CollaborationUserState.NOT_INITIATED:
```

Comment:
Again a second user with  NOT_INITIATED or PENDING could overwrite a declined. At this point you should probably first prove me wrong/right by writing tests to show this, then see what comes out

Reply:
I think the break should take care of it. If we get one state.DECLINED we bail and take that as the collaboration-wide status no matter how many other users exist. I'll write a test to double check though.

Reply:
Ah, so I see, but should the comment at line 147 say 'declined' rather than 'pending'?

Reply:
More of an 'and' then a rather. Changing to:
# 'not_initiated' should never coexist with other states on a collaboration

Anyone initiating edit switches themselves to "approved" and other user(s) to pending.